Plugins/NativeCamera/Source/Apple/CameraDevice.mm

```diff
@@ -679,13 +719,16 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
                 [m_impl->currentCommandBuffer commit];
             }
         });
-
+#if (__MAC_OS_X_VERSION_MIN_REQUIRED >= 140000 || __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000)
+        return CameraDimensions{m_impl->cameraDimensions.width, m_impl->cameraDimensions.height};
+#else
         // To match the web implementation if the sensor is rotated into a portrait orientation then the width and height
         // of the video should be swapped
         return m_impl->cameraTextureDelegate->VideoOrientation == AVCaptureVideoOrientationLandscapeLeft ||
             m_impl->cameraTextureDelegate->VideoOrientation == AVCaptureVideoOrientationLandscapeRight ?
             CameraDimensions{m_impl->cameraDimensions.width, m_impl->cameraDimensions.height} :
```

Comment:
I was wondering this too... we're using this code in product today, and we support all device orientations, so it seems like the current behavior is correct as far as I can tell. If that's the case, then it also seems like we can remove this condition. But also want to see of Colton remembers.

Reply:
It's been a while since I've looked at this code, but from what I recall MacOS doesn't support the notion of orientation changes. For MacOS devices the VideoOrientation stays at the default of LandscapeLeft which is the "normal" orientation.

Reply:
Sorry. I didn't notice that the inline if was returning the same value. Yes, that does seem like a copy & paste error. The dimensions should be swapped when in a portrait orientation... I'm surprised that isn't causing issues somewhere else down the line.

My previous comment was saying that I don't think you need to fork the logic for MacOS here as it should just always be considered to be in landscape orientation.

Reply:
Thanks! I've added the swap for portrait.